.github/workflows/ci.yml

```
@@ -0,0 +1,39 @@
+name: ci
+
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main]
```

I'd prefer it targets every branch, so everyone gets the benefits of build and test commands. If there are other needs for packaging the source code or perform any other specific command that should be run on PR only, or on main branch only, we can create a separate workflow file for that.

```suggestion
on:
  pull_request:
  push:
```